### ISSUE TITLE
Bad event handler templates bring the suite down.

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#2350]
E.g.:
```
[runtime]
    [[foo]]
         script = /bin/false
         [[[events]]]
               failed handler = echo %(nam)s  # "name"!
```
At runtime:
```
...
	  File "/home/vagrant/cylc.git/lib/cylc/task_events_mgr.py", line 842, in _setup_custom_event_handlers
	    "suite_url": quote(self.suite_url),
	KeyError: 'nam'
	
2017-07-07T13:44:27+10 ERROR - error caught: cleaning up before exit
2017-07-07T13:44:27+10 INFO - Suite shutting down - ERROR: 'nam'
2017-07-07T13:44:34+10 INFO - DONE
```

We should catch the exception, print a warning, and carry on.

**Comment by matthewrmshin** [comment]
I suppose we should catch this at validation?

**Comment by hjoliver** [comment]
That would be even better.

**Comment by matthewrmshin** [comment]
This will have to wait until after #2349 - we need to check the settings in `[[[meta]]]` for a full list of keys.

**Comment by matthewrmshin** [comment]
Fix to original issue proposed #2352. Validation to be done after #2349.
